commands.go

```diff
@@ -1249,6 +1257,177 @@ func (c *cmdable) SUnionStore(destination string, keys ...string) *IntCmd {
 
 //------------------------------------------------------------------------------
 
+// X represents stream message response.
+type XStream struct {
+	Stream  string
+	Messages []XMessage
+}
+
+// XStreamCheckpoint represents a query into a stream
+type XStreamCheckpoint struct {
+	Stream string
+	Checkpoint XMessageID
+}
+
+// XMessage represents messages within the stream response
+type XMessage struct {
+	ID XMessageID
+	Values map[string]interface{}
+}
+
+// XMessageID represents a message's ID within a stream.
+// It is composed of an ID (current timestamp if auto-assigned)
+// and a sequence ID. E.g. 1511833543298-0
+type XMessageID struct {
+	ID int64
+	SeqID int
```

Comment:
ID -> Utime or UnixTime
SeqID -> Seq

Reply:
You're not forced to make this a timestamp; e.g. `XADD stream 1 foo bar` is a valid command. For that reason, I don't think it makes sense to force it to be a time type